src/library/dao/cratedao.cpp

```diff
@@ -18,6 +18,32 @@ CrateDAO::~CrateDAO() {
 
 void CrateDAO::initialize() {
     qDebug() << "CrateDAO::initialize()";
+
+    //get the count to allocate HashMap
+    int tracksInCratesCount = 0;
+    QSqlQuery query(m_database);
```

Comment:
does query.size() return a valid value with sqlite? Than we can get around the second query.

If not, you need not Query if  tracksInCratesCount == 0